### DISCUSSION TITLE
Required vs optional fields in the properties

### DISCUSSION THREAD

**Discussion by xinyuluo** [komoot/photon#664] [Q&A] [answered]
Hi! I'm trying to construct address for results returned by Photon (unlike Nominatim which has an `address` field, different parts that constitute an address exist separately in `properties` in Photon). I noticed that some fields in the properties attributes exist in some results but not others. For instance, in

```
    {
      "geometry": {
        "coordinates": [
          -0.148266,
          51.5165737
        ],
        "type": "Point"
      },
      "type": "Feature",
      "properties": {
        "osm_id": 3524361499,
        "country": "United Kingdom",
        "countrycode": "GB",
        "postcode": "W1U 2BP",
        "locality": "East Marylebone",
        "county": "Greater London",
        "type": "house",
        "osm_type": "N",
        "osm_key": "amenity",
        "street": "Wigmore Street",
        "district": "Mayfair",
        "osm_value": "restaurant",
        "name": "Wigmore Hall Restaurant",
        "state": "England"
      }
    }
```
We have values for `country`, `locality`, `county`, `street`, `district`, and `state`, whereas for
```
    {
      "geometry": {
        "coordinates": [
          4.8048901,
          51.5606887
        ],
        "type": "Point"
      },
      "type": "Feature",
      "properties": {
        "osm_id": 339753931,
        "country": "Netherlands",
        "city": "Breda",
        "countrycode": "NL",
        "postcode": "4834SP",
        "type": "house",
        "osm_type": "N",
        "osm_key": "amenity",
        "housenumber": "100",
        "street": "Wolfslaardreef",
        "osm_value": "restaurant",
        "name": "Wolfslaar Restaurant",
        "state": "North Brabant"
      }
    }
```
we have values for `country`, `city`, `house number`, `street`, and `state`.

Is there a way to know what are the possible fields returned in properties (related to address)? Alternatively, any tips on constructing address from results returned by Photon API would be appreciated too. 

Thank you.

**Comment by lonvia** [comment, marked as answer]
See https://github.com/geocoders/geocodejson-spec/tree/master/draft

**Reply by xinyuluo** [reply]
Thank you!

**Reply by xinyuluo** [reply]
@lonvia I wonder if this link is up-to-date? Asking because "geocoding" is a required field for main object in this document (https://github.com/geocoders/geocodejson-spec/tree/master/draft#main-object); however, I saw that neither Nominatim nor Photon returns this field in their responses.

**Reply by mtmail** [reply]
Nominatim will return the field when requesting format=geocodejson https://nominatim.openstreetmap.org/search.php?q=berlin&format=geocodejson&addressdetails=1
